docs/src/main/sphinx/object-storage/file-system-s3.md

```diff
@@ -123,3 +123,138 @@ and secret keys, STS, or an IAM role:
+  of those roles.
+- `kmsKeyId`: ID of KMS-managed key to be used for client-side encryption.
+- `allowedKmsKeyIds`: KMS-managed key IDs that are allowed to be specified as an extra
+  credential. If list cotains `*`, then any key can be specified via extra credential.
+- `endpoint`: The S3 storage endpoint server. This optional property can be used
+  to override S3 endpoints on a per-bucket basis.
+- `region`: The region S3 client should connect to. This optional property can be used
+  to override S3 regions on a per-bucket basis.
+
+The security mapping entries are processed in the order listed in the configuration
+JSON. More specific mappings should thus be specified before less specific mappings.
+For example, the mapping list might have URL prefix `s3://abc/xyz/` followed by
+`s3://abc/` to allow different configuration for a specific path within a bucket
+than for other paths within the bucket. You can specify the default configuration
+by not including any match criteria for the last entry in the list.
+
+In addition to the rules above, the default mapping can contain the optional
+`useClusterDefault` boolean property set to true to use the default S3 configuration.
+It cannot be used with any other configuration settings.
+
+If no mapping entry matches and no default is configured, access is denied.
+
+The configuration JSON can either be retrieved from a file via
+`s3.security-mapping.config-file` or from a REST endpoint via
+`s3.security-mapping.config-uri`.
+
+Example JSON configuration:
+
+```json
+{
+  "mappings": [
+    {
+      "prefix": "s3://bucket-name/abc/",
+      "iamRole": "arn:aws:iam::123456789101:role/test_path"
+    },
+    {
+      "user": "bob|charlie",
+      "iamRole": "arn:aws:iam::123456789101:role/test_default",
+      "allowedIamRoles": [
+        "arn:aws:iam::123456789101:role/test1",
+        "arn:aws:iam::123456789101:role/test2",
+        "arn:aws:iam::123456789101:role/test3"
+      ]
+    },
+    {
+      "prefix": "s3://special-bucket/",
+      "accessKey": "AKIAxxxaccess",
+      "secretKey": "iXbXxxxsecret"
+    },
+    {
+      "prefix": "s3://regional-bucket/",
+      "iamRole": "arn:aws:iam::123456789101:role/regional-user",
+      "endpoint": "https://bucket.vpce-1a2b3c4d-5e6f.s3.us-east-1.vpce.amazonaws.com",
+      "region": "us-east-1"
+    },
+    {
+      "prefix": "s3://encrypted-bucket/",
+      "kmsKeyId": "kmsKey_10"
+    },
+    {
+      "user": "test.*",
+      "iamRole": "arn:aws:iam::123456789101:role/test_users"
+    },
+    {
+      "group": "finance",
+      "iamRole": "arn:aws:iam::123456789101:role/finance_users"
+    },
+    {
+      "iamRole": "arn:aws:iam::123456789101:role/default"
+    }
+  ]
+}
+```
+
+:::{list-table}
+:header-rows: 1
+
+* - Property name
+  - Description
+* - `s3.security-mapping.enabled`
+  - Enable the security mapping feature.
+* - `s3.security-mapping.config-file`
+  - The JSON configuration file containing security mappings.
```

Comment:
Path to the JSON 

relative to ? or absolute? or both

Reply:
It can be relative or absolute. I don't think we discuss "relative to what" in every place that uses additional files, but it's relative to the server configuration directory. Do we have a place that discusses that to link to?

Reply:
Not sure we have a generic place .. maybe just go with `Absolute or relative path to the JSON configuration file.`